hooks/pre_prompt.py

```diff
@@ -0,0 +1,45 @@
+import os
+import json
+import requests
+
+def get_latest_version(package_name):
+    """Fetch the latest version of a package from PyPI."""
+    url = f"https://pypi.org/pypi/{package_name}/json"
+    response = requests.get(url)
+    if response.status_code == 200:
+        data = response.json()
+        return data["info"]["version"]
+    else:
+        raise ValueError(f"Could not fetch version for {package_name}")
+
+def main():
+    """Update cookiecutter context dynamically with current package versions."""
+    # Fetch versions for the required packages
+    packages = {
+        "vivarium": "vivarium",
+        "vivarium_public_health": "vivarium-public-health",
+        "vivarium_cluster_tools": "vivarium-cluster-tools",
+        "vivarium_inputs": "vivarium-inputs",
+        "gbd_mapping": "gbd-mapping",
+    }
```

Comment:
Nit: you could just do a string replacement of `_` with `-` rather than maintaining a dictionary, so we don't need to update this every time we add a new dependency.

Reply:
I like it better with the explicit mapping, so I'll opt to leave it for now. If it gets annoying to add new dependencies, let's change it then.

Reply:
I changed my mind, will open another PR